public/swagger/index.tsx

```diff
@@ -0,0 +1,48 @@
+import '../app/core/trustedTypePolicies';
+declare let __webpack_public_path__: string;
+declare let __webpack_nonce__: string;
+
+// Check if we are hosting files on cdn and set webpack public path
+if (window.public_cdn_path) {
+  __webpack_public_path__ = window.public_cdn_path;
+}
+
+// This is a path to the public folder without '/build'
+window.__grafana_public_path__ =
+  __webpack_public_path__.substring(0, __webpack_public_path__.lastIndexOf('build/')) || __webpack_public_path__;
+
+if (window.nonce) {
+  __webpack_nonce__ = window.nonce;
+}
+
+import 'swagger-ui-react/swagger-ui.css';
+
+import DOMPurify from 'dompurify';
+import { createRoot } from 'react-dom/client';
+
+import { Page } from './SwaggerPage';
+
+// Use dom purify for the default policy
+const tt = window.trustedTypes;
+if (tt?.createPolicy) {
```

Comment:
@KristianGrafana -- does this work?  It is kinda like https://web.dev/articles/trusted-types#default-policy but the typescript compiles

Reply:
Very nice! Indeed, this work as long as we are fine with type casting. In this case we should since TS does not support Trusted Types. 

In commit https://github.com/grafana/grafana/pull/91785/commits/92345522c3d6af989947af33efe2f6f45b4436e2 I also added a way to sanitize script URL's. Also attaching proof that this works:

![Screenshot from 2024-08-13 18-48-23](https://github.com/user-attachments/assets/7f7f57e3-44aa-41f3-83a4-e710a7a39b52)